index.Rmd

```diff
@@ -399,102 +366,108 @@ For some applications, we may only be interested in larger forest areas. We may
+
+# Reset plotting window
+par(mfrow = c(1, 1)) 
 ```
 
-We have successfully removed all *island* pixels from the forest mask using the `clump()` function. We can adjust our sieve criteria to only directly adjacent (NESW) neighbours: the so-called *Rook's Case*. To accomplish this, simply repeat the code above, but supply the argument `directions=4` when calling `clump()`.
+We have successfully removed all *island* pixels from the forest mask using the `clump()` function. We can adjust our sieve criteria to only directly adjacent (NESW) neighbours: the so-called *Rook's Case*. To accomplish this, simply repeat the code above, but supply the argument `directions = 4` when calling `patches()`.
 
 ```{r, eval=FALSE, echo=FALSE, results='hide', include=FALSE}
-## Check for igraph package and install if missing
-if(!"igraph" %in% rownames(installed.packages())){install.packages("igraph")}
-library(igraph)
-
-## Group raster cells into clumps based on the Queen's Case
-?clump
-forestclumps <- clump(formask, directions=4)
-clumpFreq <- freq(forestclumps)
-## Coerce freq table to data.frame
-clumpFreq <- as.data.frame(clumpFreq)
-excludeID <- clumpFreq$value[which(clumpFreq$count==1)]
-## make a new forest mask to be sieved
+# Group raster cells into clumps based on the Rook's Case
+forestpatches <- patches(formask, directions = 4)
+patchFreq <- freq(forestpatches)
+
+# Select the ID's to later be excluded
+excludeID <- patchFreq$value[which(patchFreq$count == 1)]
+
+# Make a new forest mask to be sieved
 formaskSieve <- formask
-## assign NA to all clumps whose IDs are found in excludeID
-formaskSieve[forestclumps %in% excludeID] <- NA
-e <- extent(c(811744.8, 812764.3, 849997.8, 850920.3))
-opar <- par(mfrow = c(1, 2))
-plot(formask, ext=e, col="dark green", legend=FALSE, main='formask')
-plot(formaskSieve, ext=e, col="dark green", legend=FALSE, main='formaskSieve')
-par(opar)
+
+# Assign NA to all patches whose IDs are found in excludeID
+formaskSieve[forestpatches %in% excludeID] <- NA
+
+# Plot again
+e <- ext(c(830000, 834000, 830000, 834000))
+par(mfrow = c(1, 2)) # allow 2 plots side-by-side
+plot(formask, ext = e, col = "dark green", legend = FALSE, main = 'formask')
+plot(formaskSieve, ext = e, col = "dark green", legend = FALSE, main = 'formaskSieve')
+
+# Reset plotting window
+par(mfrow = c(1, 1)) 
 ```
 
 We could take this approach further and apply a minimum mapping unit (MMU) to our forest mask.
 
 ```{block, type="alert alert-success"}
-> **Question 5:** How could you adjust the above sieve to remove all forest pixels with area below 0.5 hectares? Consider the fact that Landsat pixels are 30m by 30m, and that one hectare is equal to 10000m<sup>2</sup>.
+> **Question 5:** How could you adjust the above sieve to remove all forest pixels with area below 0.5 hectaresif you are using Landsat satellite data? Consider the fact that Landsat pixels are 30m by 30m, and that one hectare is equal to 10000m<sup>2</sup>.
```

Comment:
Hah, this was from when we still used Landsat for this tutorial. We should instead just ask about Sentinel-2 pixels (at the resolution we are using here; it varies per band...)